### PR TITLE
Remove rustflags for apple builds

### PR DESCRIPTION
**Pull request by lcruz99**
Until now rust flags for apple devices were being set on the GLOBAL_CONFIG dictionary. This disallow the developer to customize them as the flags on the script override the value set on the manifest file.

This commit removes this flags from the GLOBAL_CONFIG and the projects who were relying on it as default will have to set them on their manifest file instead.

### CODE REVIEW

**Comment by arg0d** [comment]
So the flags `-C link-arg=-s -C embed-bitcode` are still needed, right?

**Comment by lcruz99** [comment]
`embed-bitcode` removal is part of the planned binary optimizations and also deprecated on XCode 14. `-s` linker flag should be set per project on their manifest file, libnudler was already doing this, libdrop don't. Also, if one project eventually wants to use a different one, it would get overriden by this one.

**Comment by lcruz99** [comment]
I believe is the responsability of a project when upgrading this project tag that the changelog is read.

**Comment by lcruz99** [comment]
(Updated `CHANGELOG.md`)

**Comment by lcruz99** [comment]
(Updated CHANGELOG.md)

**Comment by jjanowsk** [comment]
I'm fine with those being removed from global config. Just please update the entry in CHANGELOG from UNRELEASED to v2.0.0 and create a 2.0.0 tag once this PR is merged. There is no need to do it in a separate PR. Also maintainers of libnudler, libdrop and possibly moose should be informed. Please add them to this review

**Comment by lcruz99** [comment]
@NordSecurity/libdrop-team @nymoral

**Comment by arg0d** [comment]
I feel like this type of change requires a more extensive explanation in changelog. `Remove RUSTFLAGS for apple builds` will not mean anything to a developer who will be updating to the new version. I would also include the changes required to the 
consumers of this repository, e.g.

## 2.0.0
- **BREAKING**: Remove `RUSTFLAGS` for apple builds. The removal of `RUSTFLAGS` allows the consuming projects to configure `RUSTFLAGS` independently. Consumers must set `RUSTFLAGS="-C link-arg=-s -C embed-bitcode"` after updating to this version.

**Comment by lcruz99** [comment]
done @arg0d @jjanowsk

**Comment by arg0d** [comment]
+2

**Review by arg0d — APPROVED** [review]

**Pull request merged**
